### ISSUE TITLE
Can't set NoNewPrivileges=false and RestrictSUIDSGID=false for DynamicUser=true

### ISSUE THREAD

**Issue by lilydjwg** [systemd/systemd#12476]
**systemd version the issue has been seen with**
> 242

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > The following config will take effect

```
[Service]
User=someuser
Group=somegroup
DynamicUser=true
NoNewPrivileges=false
RestrictSUIDSGID=false
```

**Unexpected behaviour you saw**
 > The `NoNewPrivileges=false` and `RestrictSUIDSGID=false` don't take effect.

**Steps to reproduce the problem**
* find a cap_net_bind_service,cap_net_admin+ep binary and try to run it with `DynamicUser=true`
* the service won't get the needed caps

I find the `DynamicUser` option a much easier way to make different services use seperate users instead of using `nobody` together, or creating a user for each service. But some service does need some privileges to run. I prefer to add a drop-in config for those `nobody` services rather than report upstream and wait forever for them to create a user for each service. With systemd 242 I can no longer apply this simple security improvement for some services.

**Comment by poettering** [comment]
We locked this down, since there are some (mostly theoretical) ways to create a suid binary under the dynamic user id and have it stick around when we don't turn off suid handling. And we can't have that, since that means the dynamic uids wouldn't be so dynamic after all. I fear if you need suid tools then dynamic uids are not a mechanism you can make use of, sorry...

**Comment by lilydjwg** [comment]
OK, sorry to hear that but thanks for the explanation. Could you make this clear in the docs? It says "implicitly enabled" but doesn't say they can't be disabled.

**Comment by poettering** [comment]
ok, let's reopen this and turn this into a documentation bug, to document what @lilydjwg suggested explicitly

**Comment by boucman** [comment]
@poettering it seems you forgot to actually reopen the bug...

**Comment by poettering** [comment]
indeed! done!

**Comment by poettering** [comment]
fix waiting in #12868
